### ISSUE TITLE
Got ConcurrentModificationException

### ISSUE THREAD

**Issue by PeronGH** [acsbendi/Android-Request-Inspector-WebView#3]
I tried to use OkHttp to provide fake response for all requests:

```kotlin
webViewClient = object : RequestInspectorWebViewClient(this@apply) {

    private val httpClient = OkHttpClient()

    @RequiresApi(Build.VERSION_CODES.TIRAMISU)
    override fun shouldInterceptRequest(
        view: WebView,
        webViewRequest: WebViewRequest
    ): WebResourceResponse {

        val request = Request.Builder()
            .url(BASE_URL + webViewRequest.url)
            .method(
                webViewRequest.method,
                (if (HttpMethod.permitsRequestBody(webViewRequest.method)) webViewRequest.body else null
                    ?.toRequestBody(webViewRequest.headers["content-type"]?.toMediaTypeOrNull())
            )
            .headers(webViewRequest.headers.toHeaders())
            .build()

        val response = try {
            httpClient.newCall(request).execute()
        } catch (e: Exception) {
            Log.e("WebView", "Request failed at " + request.url)
            return WebResourceResponse(
                "text/plain",
                "utf-8",
                InputStream.nullInputStream()
            )
        }

        Log.d(
            "WebView",
            "${response.protocol} ${response.request.method} ${response.request.url}"
        )

        val type = response.headers["content-type"]?.toMediaTypeOrNull()
        return WebResourceResponse(
            if (type != null)
                "${type.type}/${type.subtype}"
            else null,
            response.headers["content-encoding"] ?: "utf-8",
            response.code,
            response.message.ifEmpty { getDefaultMessage(response.code) },
            response.headers.toMultimap().mapValues { it.value.first() },
            response.body?.byteStream()
        )
    }
}
```

But I got this error from time to time:

```
FATAL EXCEPTION: main
Process: com.example.browser, PID: 14726
java.util.ConcurrentModificationException
	at java.util.ArrayList$Itr.next(ArrayList.java:860)
	at com.acsbendi.requestinspectorwebview.RequestInspectorJavaScriptInterface.findRecordedRequestForUrl(RequestInspectorJavaScriptInterface.kt:21)
	at com.acsbendi.requestinspectorwebview.RequestInspectorWebViewClient.shouldInterceptRequest(RequestInspectorWebViewClient.kt:29)
	at K9.a(chromium-TrichromeWebViewGoogle6432.aab-beta-535902244:80)
	at org.chromium.android_webview.AwContentsBackgroundThreadClient.shouldInterceptRequestFromNative(chromium-TrichromeWebViewGoogle6432.aab-beta-535902244:15)
```

**Comment by moesaldi** [comment]
is there any solution for this issue?

**Comment by acsbendi** [comment]
@PeronGH @moesaldi This should be fixed in the latest version (1.0.3).
